app/models/driver.rb

```diff
@@ -0,0 +1,38 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+
+  validates :name, presence: true, format: {with: /[a-zA-Z]/}
+  validates :vin, presence: true
+
+  def total_earning
+    trips = Trip.where(driver_id: id)
+    sum = 0.00
+    trips.each do |trip|
+      sum += trip.cost
+    end
+    sum = sum * 0.85
+    return sum
+  end
+
+  def average_rating
+    ratings = Trip.where(driver_id: id)
```

Comment:
When this method is called, you have access to a specific driver instance. Any Driver instance should already have a "dot method" which will give you access to the trips for that driver. You should be using this functionality instead of reaching out from the `Trip` object.